kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/RawCustomResourceOperationsImpl.java

```diff
@@ -88,12 +91,12 @@
   private final String deletionPropagation;
   private final boolean cascading;
 
-  private enum HttpCallMethod { GET, POST, PUT, DELETE }
+  private enum HttpCallMethod { GET, POST, PUT, PATCH, DELETE }
 
   private RawCustomResourceOperationsImpl(OkHttpClient client, Config config, CustomResourceDefinitionContext crdContext, String namespace, String name, long gracePeriodInSeconds, boolean cascading, String deletionPropagation, ListOptions listOptions, boolean dryRun) {
     super(client, config);
     this.customResourceDefinition = crdContext;
-    this.objectMapper = Serialization.jsonMapper();
+    this.objectMapper = new ObjectMapper();
```

Comment:
Why can't we reuse the global ObjectMapper singleton instance?

Reply:
You can test from my example:
https://github.com/andreaTP/fabric8-edit-as-patch/blob/8035f945a3c0be2db239c92632dc91bb4dda6d9e/src/main/scala/Main.scala#L59

After registering `DefaultScalaModule` the global ObjectMapper will deserialize as scala Maps resulting in class cast exception e.g. here:
https://github.com/fabric8io/kubernetes-client/blob/e60c6348bb3730b2c909aee4f71cbf5db86cc3f7/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/RawCustomResourceOperationsImpl.java#L870

Reply:
OK, that's very specific for your use case, and I'm sure you are having issues in other places around the code-base.

We should really need to find a neater way to address the problem.

Especially because if we start creating ObjectMapper instances across the different service implementations we'll have more problems in the future.

Reply:
Please just propose me a solution, I will go-ahead 🙂 
Here we _RELY_ on having a clean `ObjectMapper`, but, as said I'm open to suggestions!

Just to mention, no, fortunately, I have no other Serialization issues:
 - [Cloudflow Operator](https://github.com/lightbend/cloudflow/tree/master/tools/cloudflow-operator/src/main/scala/cloudflow/operator) handling 3 different CR, one CRD, ConfigMaps, Secrets, Volumes etc. etc.
 - [Cloudflow CLI](https://github.com/lightbend/cloudflow/blob/master/tools/cloudflow-cli/src/main/scala/akka/cli/cloudflow/kubeclient/KubeClientFabric8.scala) reading and writing again CRs, Secrets and more

Reply:
@manusa any suggestion to unblock this PR?
should we use the `PatchUtils` `ObjectMapper`? https://github.com/fabric8io/kubernetes-client/blob/3ed4809e50f2e71a362487318331ae66bf644931/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/PatchUtils.java#L40

I still think that a fresh one is more appropriate here until we have a more "structural" solution.
WDYT?

Reply:
Yes, that seems very reasonable.